Source/WebCore/page/EventHandler.cpp

```diff
@@ -2642,21 +2643,44 @@ void EventHandler::cancelDragAndDrop(const PlatformMouseEvent& event, std::uniqu
     clearDragState();
 }
 
-bool EventHandler::performDragAndDrop(const PlatformMouseEvent& event, std::unique_ptr<Pasteboard>&& pasteboard, OptionSet<DragOperation> sourceOperationMask, bool draggingFiles)
+void EventHandler::performDragAndDrop(const PlatformMouseEvent& event, std::unique_ptr<Pasteboard>&& pasteboard, OptionSet<DragOperation> sourceOperationMask, bool draggingFiles, const HitTestResult& result, CompletionHandler<void(bool)>&& completionHandler)
 {
     Ref frame = m_frame.get();
 
     bool preventedDefault = false;
+    auto subframe = EventHandler::subframeForTargetNode(result.protectedTargetNode().get());
+    auto afterDragAndDrop = [weakThis = WeakPtr { *this }, this, completionHandler = WTFMove(completionHandler)](bool preventedDefault) mutable {
+        if (weakThis)
+            clearDragState();
+        completionHandler(preventedDefault);
+    };
+
+#if PLATFORM(COCOA) && ENABLE(DRAG_SUPPORT)
+    RefPtr remoteFrame = dynamicDowncast<RemoteFrame>(subframe).get();
+    if (remoteFrame) {
+        if (auto remoteUserInputEventData = userInputEventDataForRemoteFrame(remoteFrame.get(), result.roundedPointInInnerNodeFrame())) {
+            remoteFrame->client().propagateDragAndDrop(frame->frameID(), *remoteUserInputEventData, WTFMove(afterDragAndDrop));
```

Comment:
To me this seems like a weird place to do this, I don't think other places in WebCore send and expect a reply from a frame hosted in another web process like this.

Reply:
In `Web/RemoteFrameClient`, There's `bindRemoteAccessibilityFrames` for precedent, though it's sync. `documentURLForConsoleLog` is async, but doesn't contact another frame in WPP.